src/preprocessing/passes/non_clausal_simp.cpp

```diff
@@ -448,10 +431,106 @@ PreprocessingPassResult NonClausalSimp::applyInternal(
         Rewriter::rewrite(Node(learnedBuilder)));
   }
 
-  propagator->setNeedsFinish(true);
   return PreprocessingPassResult::NO_CONFLICT;
 }  // namespace passes
 
+std::pair<bool, std::vector<Node>> NonClausalSimp::preprocessByCryptoMinisat(
+    AssertionPipeline* assertionsToPreprocess, unsigned substs_index)
+{
+  std::unique_ptr<SatSolver> satSolver(SatSolverFactory::createCryptoMinisat(
+      smtStatisticsRegistry(), "non_clausal_simp_solver"));
+  CVC4::prop::NullRegistrar registrar;
+  context::Context context;
+  CVC4::prop::TseitinCnfStream cnfStream(satSolver.get(),
+                                         &registrar,
+                                         &context,
+                                         true,
+                                         "toCNF-skeleton-preprocessing");
+  std::vector<Node> learned_literals;
+  Trace("skeleton-preprocessing") << "asserting to sat solver" << std::endl;
```

Comment:
Right now, the trace tags are inconsistent. I'd just use `non-clausal-simplify` for everything since it is all in the same file.